### ISSUE TITLE
Not able to document `@resumable` functions

### ISSUE THREAD

**Issue by siddharthlal25** [JuliaDynamics/ResumableFunctions.jl#45]
Doing this:
```
"""
    filenames(...)

docstring
"""
@resumable function filenames(...)
...
end
```
throws an error, error being:
```
cannot document the following expression:

#= /home/siddharth/.julia/dev/CCDReduction.jl/src/collection.jl:105 =# @resumable function filenames(df::DataFrame)
```

**Comment by siddharthlal25** [comment]
The workaround was:
```
"""
    filenames(...)

docstring
"""
function filenames end

@resumable function filenames(...)
...
end
```

It would be nice to add a functionality that allows users to document directly above `@resumable` function.

**Comment by hdavid16** [comment]
Yes, this would be nice

**Comment by BenLauwens** [comment]
I will look at this issue but I think the workaround is the easiest way to document resumable functions

**Comment by detrin** [comment]
I am using Documenter.jl and I have similar issue. The workaround is in my opinion only temporal solution. I suggest adding encapsulation of the function as follows
```julia
@resumable begin
"""
    filenames(...)

docstring
"""
function filenames(...)
...
end
end
```

**Comment by Krastanov** [comment]
To be fixed in #71 - for future reference, one simply needs to use `Base.@__doc__` for situations like this.
